starlite/middleware/session/base.py

```diff
@@ -0,0 +1,313 @@
+
+            ```python
+            from os import urandom
+
+            from starlite import Starlite, Request, get
+            from starlite.middleware.session import SessionCookieConfig
+
+            session_config = SessionCookieConfig(secret=urandom(16))
+
+
+            @get("/")
+            def my_handler(request: Request) -> None:
+                ...
+
+
+            app = Starlite(route_handlers=[my_handler], middleware=[session_config.middleware])
+            ```
+
+        Returns:
+            An instance of DefineMiddleware including 'self' as the config kwarg value.
+        """
+        return DefineMiddleware(SessionMiddleware, backend=self._backend_class(config=self))
+
+
+class ServerSideSessionConfig(BaseBackendConfig):
+    session_id_bytes: int = 32
+    """
+    Number of bytes used to generate a random session-ID
+    """
+
+
+ConfigT = TypeVar("ConfigT", bound=BaseBackendConfig)
+ServerConfigT = TypeVar("ServerConfigT", bound=ServerSideSessionConfig)
+
+
+class SessionBackend(abc.ABC, Generic[ConfigT]):
+    def __init__(self, config: ConfigT) -> None:
+        """Generic session backend, for either client- or server-side
+        storage."""
+        self.config = config
+
+    @staticmethod
+    def serialise_data(data: "ScopeSession", scope: Optional["Scope"] = None) -> bytes:
+        """Serialise data into bytes for storage in the backend.
+
+        The serialiser will be
+        gathered from `scope` and fall back to [default_serializer][starlite.utils.default_serializer]
+        """
+        serializer = (get_serializer_from_scope(scope) if scope else None) or default_serializer
+        return dumps(data, default=serializer, option=OPT_SERIALIZE_NUMPY)
+
+    @staticmethod
+    def deserialise_data(data: Any) -> Dict[str, Any]:
+        """Deserialise data into a dictionary for use in the application
+        scope."""
+        return cast("Dict[str, Any]", loads(data))
+
+    @abc.abstractmethod
+    async def store_in_message(
+        self, scope_session: "ScopeSession", message: "Message", connection: ASGIConnection
+    ) -> None:
+        """Stores the necessary information in the outgoing `Message`
+
+        Args:
+            scope_session: Current session to store
+            message: Outgoing send-message
+            connection: Originating ASGIConnection containing the scope
+        """
+
+    @abc.abstractmethod
+    async def load_from_connection(self, connection: ASGIConnection) -> Dict[str, Any]:
+        """Load session data from a connection and return it as a dictionary to
+        be used in the current application scope.
+
+        Args:
+            connection: Originating ASGIConnection
+
+        Notes:
+            - This should not modify the connection's scope. The data returned by this
+            method will be stored in the application scope by the middleware
+        """
+
+
+class ServerSideBackend(Generic[ServerConfigT], SessionBackend[ServerConfigT], abc.ABC):
```

Comment:
I don't think you need to explicitly inherit from `ABC` here, since you already inherit from `SessionBackend` which inherits from ABC.

Reply:
note- this is also fine as it is, for readability.

Reply:
Somehow I had pyright complaining about the mro without an explicit `ABC`. Will try again to see if it's resolvable

Reply:
Done. Not sure what I did there earlier.